CMakeLists.txt

```diff
@@ -37,7 +37,8 @@ set(${PROJECT_NAME}_SRCS
   src/meta_object.cpp
   src/multi_library_class_loader.cpp
 )
-add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS})
+file(GLOB ${PROJECT_NAME}_HDRS "include/${PROJECT_NAME}/*.h") # allow headers to be shown in CodeBlocks/QtCreator
```

Comment:
I know. However, finding the headers is not functionally relevant here (dependency analysis will be performed by make).
It only helps some IDEs like CodeBlocks and QtCreator to list all headers of the project too. And for that purpose globbing is perfect. One doesn't need to care about it anymore ;-)